superbench/analyzer/data_diagnosis.py

```diff
@@ -169,7 +169,11 @@ def _run_diagnosis_rules_for_single_node(self, node):
                 violated_num = rule_op(data_row, self._sb_rules[rule], summary_data_row, details, categories)
             # label the node as defective one
             if self._sb_rules[rule]['store']:
-                violation[rule] = violated_num
+                store_values[rule] = {}
+                store_values[rule]['metrics'] = {}
+                for metric in self._sb_rules[rule]['metrics']:
+                    store_values[rule]['metrics'][metric] = data_row[metric]
+                store_values[rule]['violated'] = violated_num
```

Comment:
Big concern to change original features. It mean that the file is not back-compitable. Why do we support in this way?
I think we should change the `store` part and `label["rule1"]` still store the violation results if you have criteria

Reply:
do you mean if store is True and 'criteria' is None, we store the values of the metrics, elif store is Ture and 'criteria' is not None, we still store the violation results of the rule?